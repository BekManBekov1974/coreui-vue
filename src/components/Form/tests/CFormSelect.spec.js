import { mount } from '@vue/test-utils'
import Component from '../CFormSelect'

const ComponentName = 'CFormSelect'
const customWrapper = mount(Component, {
  propsData: {
    label: 'label',
    id: 'some_id',
    options: ['Option 1', 'Option 2', 'Option 3'],
    placeholder: 'placeholder',
    wasValidated: true,
    checked: true,
    value: 'value',
    validFeedback: 'input is valid',
    invalidFeedback: 'input is invalid',
    tooltipFeedback: true,
    description: 'permission',
    isValid: true,
    addInputClasses: 'additional-input-class',
    addLabelClasses: 'additional-label-class',
    custom: true,
    inline: true,
    size: 'lg',
    appendHtml: 'appended',
    prependHtml: 'prepended'
  }
})

describe(ComponentName, () => {
  it('has a name', () => {
    expect(Component.name).toMatch(ComponentName)
  })
  it('renders correctly', () => {
    expect(customWrapper.element).toMatchSnapshot()
  })
})
