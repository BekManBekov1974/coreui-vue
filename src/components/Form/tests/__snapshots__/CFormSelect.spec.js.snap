// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CFormSelect renders correctly 1`] = `
<div
  class="c-form-group"
  role="group"
>
   
  <!---->
   
  <select
    class="c-form-control-plaintext c-form-control-lg"
    id="some_id"
  >
    <option
      disabled="disabled"
      hidden="hidden"
      selected="selected"
      value=""
    >
      
        placeholder
      
    </option>
     
    <option
      data-key="0"
      value="Option 1"
    >
      
          Option 1
        
    </option>
    <option
      data-key="1"
      value="Option 2"
    >
      
          Option 2
        
    </option>
    <option
      data-key="2"
      value="Option 3 "
    >
      
          Option 3 
        
    </option>
  </select>
    
  <!---->
   
  <!---->
   
  <!---->
   
  <!---->
</div>
`;

exports[`CFormSelect renders correctly 2`] = `
<div
  class="c-form-group"
  role="group"
>
   
  <!---->
   
  <select
    class="c-custom-select c-custom-select-lg"
    id="some_id"
  >
    <!---->
     
    <option
      data-key="0"
      value="Option 1"
    >
      
          Option 1
        
    </option>
    <option
      data-key="1"
      value="Option 2"
    >
      
          Option 2
        
    </option>
    <option
      data-key="2"
      value="Option 3 "
    >
      
          Option 3 
        
    </option>
  </select>
    
  <!---->
   
  <!---->
   
  <!---->
   
  <!---->
</div>
`;

exports[`CFormSelect renders correctly 3`] = `
<div
  class="c-form-group c-was-validated"
  role="group"
>
  <label
    class="additional-label-class c-col-form-label-lg"
    for="some_id"
  >
    
        label
      
  </label>
   
  <div
    class="c-input-group c-input-group-lg"
  >
    <div
      class="c-input-group-prepend"
    >
      <span
        class="c-input-group-text"
      >
        prepended
      </span>
    </div>
     
    <select
      checked="checked"
      class="c-custom-select c-is-valid additional-input-class"
      id="some_id"
      inline="true"
    >
      <option
        disabled="disabled"
        hidden="hidden"
        selected="selected"
        value=""
      >
        
        placeholder
      
      </option>
       
      <option
        data-key="0"
        value="Option 1"
      >
        
          Option 1
        
      </option>
      <option
        data-key="1"
        value="Option 2"
      >
        
          Option 2
        
      </option>
      <option
        data-key="2"
        value="Option 3 "
      >
        
          Option 3 
        
      </option>
    </select>
      
    <div
      class="c-input-group-append"
    >
      <span
        class="c-input-group-text"
      >
        appended
      </span>
    </div>
     
    <div
      class="c-valid-tooltip"
    >
      input is valid
    </div>
     
    <div
      class="c-invalid-tooltip"
    >
      input is invalid
    </div>
     
    <small
      class="c-form-text c-text-muted c-w-100"
    >
      permission
    </small>
  </div>
</div>
`;
